src/Vertica.Umbraco.Headless.Core/Extensions/EnumerableExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Vertica.Umbraco.Headless.Core.Extensions
+{
+    public static class EnumerableExtensions
+    {
+        public static async Task<TResult[]> ToArrayAsync<T, TResult>(this IEnumerable<T> collection, Func<T, Task<TResult>> func)
+        {
+            var list = new List<TResult>();
+            foreach (var item in collection)
+            {
+                var result = await func(item);
+                list.Add(result);
+            }
+            return list.ToArray();
```

Comment:
We should probably hand this a CancellationToken as well.
```suggestion
            return await Task.WhenAll(collection.Select(func));
```

Reply:
This would execute the tasks in parallel, would it not? I went for iterative approach to try not breaking any implementations here.

Reply:
Correct. You are right not to change it then.